deployer/src/deployment/run.rs

```diff
@@ -269,43 +277,191 @@ impl Built {
+                        ",
+                        shuttle_resource.r#type,
+                        RESOURCE_SCHEMA_VERSION,
+                        shuttle_resource.version
+                    );
+                }
+
+                // TODO?: Make the version integer be part of the cached output
+
+                match shuttle_resource.r#type {
+                    resource::Type::Database(db_type) => {
+                        // no config fields are used yet, but verify the format anyways
+                        let _config: DbInput =
+                            serde_json::from_value(shuttle_resource.config.clone())
+                                .context("deserializing resource config")?;
+
+                        let output = get_cached_output_or_provision!(DatabaseResource, {
+                            let mut req = Request::new(DatabaseRequest {
+                                project_name: service_name.to_string(),
+                                db_type: Some(db_type.into()),
+                                // other relevant config fields would go here
+                            });
+                            req.extensions_mut().insert(claim.clone());
+                            let res = provisioner_client
+                                .provision_database(req)
+                                .await?
+                                .into_inner();
+                            DatabaseResource::Info(res.into())
+                        });
+
+                        resources_to_save.push(record_request::Resource {
+                            r#type: shuttle_resource.r#type.to_string(),
+                            // Send only the config fields that affect provisioning
+                            // For now, this is "null" for all database types
+                            config: serde_json::to_vec(&serde_json::Value::Null).unwrap(),
+                            data: serde_json::to_vec(&output).unwrap(),
+                        });
+                        *r = serde_json::to_vec(&ShuttleResourceOutput {
+                            output,
+                            custom: shuttle_resource.custom,
+                        })
+                        .unwrap();
+                    }
+                    resource::Type::Secrets => {
+                        resources_to_save.push(record_request::Resource {
+                            r#type: shuttle_resource.r#type.to_string(),
+                            config: serde_json::to_vec(&serde_json::Value::Null).unwrap(),
+                            data: serde_json::to_vec(&new_secrets).unwrap(),
+                        });
+                        *r = serde_json::to_vec(&ShuttleResourceOutput {
+                            output: new_secrets.clone(),
+                            custom: shuttle_resource.custom,
+                        })
+                        .unwrap();
```

Comment:
Same comment as earlier about each loop iteration doing too much. It should be possible to simplify this.